### PR TITLE
Support token-level loss, make default

### PR DESCRIPTION
**Pull request by tyler-griggs**
## What does this PR do?
Adds support for token-level loss (ie, `token_mean` loss reduction type) as introduced by DAPO. 

With `token_mean` loss reduction, all tokens in all sequences contribute equally to loss. 

The loss reduction type is configurable via `trainer.algorithm.loss_reduction`, but the default is updated to be `token_mean`, as opposed to our previous implementation (`sequence_mean`). This loss reduction is what the community is standardizing on as default (TRL's [default](https://github.com/huggingface/trl/pull/2881), verl's [default](https://github.com/volcengine/verl/blob/517cc23c9dbb0da5c2cd2b012466790e29cb781a/verl/trainer/config/actor/actor.yaml#L63)) 

Wandb report of comparing `token_mean` vs `sequence_mean`: https://wandb.ai/sky-posttraining-uc-berkeley/gsm8k/reports/Token-level-loss-token_mean---VmlldzoxMzYwMDc4MQ

The only plot with a notable difference is `policy_loss`, which is much larger for `token_mean` than it is for `sequence_mean`:
Screenshot 2025-07-15 at 9 52 57 AM

However, this `policy_loss` matches the same magnitude of `pg_loss` we observe in verl:
Screenshot 2025-07-15 at 9 54 39 AM

### CODE REVIEW

File: skyrl-train/tests/cpu/algorithms/test_losses.py [hunk 1]
```diff
@@ -44,3 +44,109 @@ def test_policy_loss_dual_clip():
...
+            [2.0, 2.0, 2.0],  # sequence 1: consistently higher advantages
+            [1.0, 1.0, 1.0],  # sequence 2: consistently lower advantages
+        ],
+        device=device,
+    )
+
+    old_log_probs = torch.tensor([[-1.0, -1.0, -1.0], [-1.0, -1.0, -1.0]], device=device)
+
+    log_probs = torch.tensor(
+        [[-1.5, -0.5, -1.2], [-0.8, -1.3, -0.9]],  # ratios ≈ [0.61, 1.65, 0.83]  # ratios ≈ [1.22, 0.74, 1.11]
```
Comment (SumanthRH) [inline skyrl-train/tests/cpu/algorithms/test_losses.py]:
```suggestion
        [[-1.5, -0.5, -1.2], [-0.8, -1.3, -0.9]],  # ratios ≈ [[0.61, 1.65, 0.83],[1.22, 0.74, 1.11]]
```

File: skyrl-train/tests/cpu/algorithms/test_losses.py [hunk 2]
```diff
@@ -44,3 +44,109 @@ def test_policy_loss_dual_clip():
...
+        device=device,
+    )
+
+    # Create masks to test sequences with different numbers of valid tokens
+    loss_mask = torch.tensor([[1.0, 1.0, 1.0], [1.0, 0.0, 0.0]], device=device)
+
+    # Test token_mean without mask
+    loss_fn_token = PolicyLoss(loss_type="regular", loss_reduction="token_mean")
```
Comment (SumanthRH) [inline skyrl-train/tests/cpu/algorithms/test_losses.py]: nit: you should explicitly pass in the eps low and eps high values here to make the test less brittle

**Review by SumanthRH — APPROVED** [review]
Great! 

Left a couple nits

**Comment by SumanthRH** [comment]
Would be nice to add a screenshot for convergence on gms8k (and how it changes from before) before merging

**Comment by tyler-griggs** [comment]
Added details form gsm8k run in initial PR description.
